app/src/features/apiClient/screens/apiClient/utils.ts

```diff
@@ -410,3 +411,71 @@ export const filterRecordsBySearch = (records: RQAPI.Record[], searchValue: stri
 
   return records.filter((record) => matchingRecords.has(record.id));
 };
+
+export const clearActiveKeysFromSession = (keysToBeDeleted: string[]) => {
+  const activeKeys = sessionStorage.getItem(SESSION_STORAGE_ACTIVE_COLLECTIONS_KEY, []);
+
+  if (keysToBeDeleted.length === 0) {
+    return;
+  }
+
+  const updatedActiveKeys: string[] = [];
+
+  activeKeys.forEach((key: string) => {
+    if (!keysToBeDeleted.includes(key)) updatedActiveKeys.push(key);
+  });
+
+  sessionStorage.setItem(SESSION_STORAGE_ACTIVE_COLLECTIONS_KEY, updatedActiveKeys);
+};
```

Comment:
```suggestion
 if (keysToBeDeleted.length === 0) {
    return;
  }
 
  const activeKeys = sessionStorage.getItem(SESSION_STORAGE_ACTIVE_COLLECTIONS_KEY, []);
  
  const updatedActiveKeys: string[] = [];

  activeKeys.forEach((key: string) => {
    if (!keysToBeDeleted.includes(key)) updatedActiveKeys.push(key);
  });

  sessionStorage.setItem(SESSION_STORAGE_ACTIVE_COLLECTIONS_KEY, updatedActiveKeys);
};
```

Reply:
Have a util to get the expandedCollectionIds from session storage and reuse that, similarly for updation have a util.

```suggestion
export const getExpandedCollectionIdsFromSessionStorage = () => {
return sessionStorage.getItem(SESSION_STORAGE_ACTIVE_COLLECTIONS_KEY, []);
}
```